modules/apps/transfer/client/cli/tx.go

```diff
@@ -111,7 +117,7 @@ corresponding to the counterparty channel. Any timeout set to 0 is disabled.`),
 			}
 
 			msg := types.NewMsgTransfer(
-				srcPort, srcChannel, coin, sender, receiver, timeoutHeight, timeoutTimestamp,
+				srcPort, srcChannel, coin, sender, receiver, timeoutHeight, timeoutTimestamp, []byte(metadataStr),
```

Comment:
Would it be possible to pass in as metadata some serialized protocol buffer data, for example? Or is it expected that the metadata should only be string information?

Reply:
Maybe it should be bytes on those functions too. That give users more flexibility. I can make the change

Reply:
Thanks in advance! Yes, I think that generalization will be good to have.

Reply:
ohh wait, looked at this in detail now. We're already using bytes everywhere else, and I think it makes more sense to keep it as string in the CLI so that it's easy to write/use. We could use hex encoded bytes, but for 99% of the cases we'd end up making it harder for the users. 

I think if they really need bytes, they can create the message manually and pass it via grpc.